compiler/circle-input-names/src/circle-input-names.cpp

```diff
@@ -0,0 +1,194 @@
+    auto node_ = dynamic_cast<CLASS *>(node); \
+    node_->padding(Padding::SAME);            \
+    break;                                    \
+  }
+    CIRCLE_NODE(AVERAGE_POOL_2D, CircleAveragePool2D)
+    CIRCLE_NODE(CONV_2D, CircleConv2D)
+    CIRCLE_NODE(DEPTHWISE_CONV_2D, CircleDepthwiseConv2D)
+    CIRCLE_NODE(L2_POOL_2D, CircleL2Pool2D)
+    CIRCLE_NODE(MAX_POOL_2D, CircleMaxPool2D)
+    CIRCLE_NODE(TRANSPOSE_CONV, CircleTransposeConv)
+#undef CIRCLE_NODE
+    default:
+    {
+      break;
+    }
+  }
+  return;
+}
+
+// Add mode option to MIRROR_PAD, ROPE nodes
+void add_mode_option(CircleNode *node)
+{
+  switch (node->opcode())
+  {
+#define CIRCLE_NODE(OPCODE, CLASS)            \
+  case luci::CircleOpcode::OPCODE:            \
+  {                                           \
+    auto node_ = dynamic_cast<CLASS *>(node); \
+    auto mode_ = node_->mode();               \
+    node_->mode(decltype(mode_)(1));          \
+    break;                                    \
+  }
+    CIRCLE_NODE(MIRROR_PAD, CircleMirrorPad)
+    CIRCLE_NODE(ROPE, CircleRoPE)
+#undef CIRCLE_NODE
+    default:
+    {
+      break;
+    }
+  }
+  return;
+}
+
+// Fill dummy values to CircleNode for creating NodeSummary
+void fill_dummies_for_summary_creation(CircleNode *node)
+{
+  add_fused_actfn_option(node);
+  add_padding_option(node);
+  add_mode_option(node);
+}
+
+// Mock Symbol Table for CircleNodeSummaryBuilder
+class MockSymbolTable : public locop::SymbolTable
+{
+  std::string lookup(const loco::Node *) const override { return ""; }
+};
+
+// Create NodeSummary using CircleNodeSummaryBuilder and MockSymbolTable
+locop::NodeSummary create_circle_node_summary(CircleNode *node)
+{
+  locop::NodeSummary s;
+  MockSymbolTable tbl;
+  CircleNodeSummaryBuilder builder;
+
+  builder.build(node, &tbl, s);
+  return s;
+}
+
+// Get input names of CircleNode and export as JSON format
+void get_input_names_from_summary(CircleNode *node, locop::NodeSummary &s,
+                                  crew::JsonExport &json_export)
+{
+  std::vector<std::string> arg_names;
+  for (int i = 0; i < node->arity(); i++)
+  {
+    auto args = s.args().at(i);
+    // args : pair(name, value)
+    arg_names.emplace_back(args.first);
+  }
+
+  // s.opname() : "Circle.Opname""
+  auto opname = s.opname().substr(7);
+  // "Opname" : ["arg1", "arg2",...],"
+  json_export.key_val(opname, arg_names, true);
+}
+
+int main(void)
+{
+  std::stringstream ss;
+  crew::JsonExport json_export(ss);
+  // "{"
+  json_export.open_brace();
+#define CIRCLE_NODE(OP, CIRCLE_OP)                             \
+  {                                                            \
+    auto node = CircleNodeCreator<CIRCLE_OP>();                \
+    fill_dummies_for_summary_creation(&node);                  \
+    auto summary = create_circle_node_summary(&node);          \
+    get_input_names_from_summary(&node, summary, json_export); \
+  }
+#define CIRCLE_VNODE(_1, _2)
```

Comment:
Just curiosity, what does this code mean? What does it do?

Reply:
Defined to ignore `CIRCLE_VNODE`s in `circle_nodes.lst`.

Reply:
Ah, I got it. This would be better to be described as a comment. Maybe in the next PR.